packages/marketplace/src/components/ui/client-app-detail/client-app-uninstall-confirmation.tsx

```diff
@@ -0,0 +1,170 @@
+import * as React from 'react'
+import { useHistory } from 'react-router'
+import { Dispatch } from 'redux'
+import { useDispatch, useSelector } from 'react-redux'
+import { AppDetailModel } from '@reapit/foundations-ts-definitions'
+import appPermissionContentStyles from '@/styles/pages/app-permission-content.scss?mod'
+import { Button, Modal } from '@reapit/elements'
+import { clientFetchAppDetail } from '@/actions/client'
+import { appInstallationsRequestUninstall } from '@/actions/app-installations'
+import CallToAction from '../call-to-action'
+import { selectClientId } from '@/selector/client'
+import { selectInstallationFormState } from '@/selector/installations'
+import routes from '@/constants/routes'
+
+export type ClientAppUninstallConfirmationProps = {
+  appDetailData?: AppDetailModel
+  visible: boolean
+  closeUninstallConfirmationModal: () => void
+}
+
+export const handleUninstallAppSuccessCallback = (
+  appId: string,
+  clientId: string,
+  dispatch: Dispatch<any>,
+  setIsSuccessAlertVisible: (isVisible: boolean) => void,
+  closeUninstallConfirmationModal: () => void,
+) => {
+  return () => {
+    dispatch(
+      clientFetchAppDetail({
+        id: appId,
+        clientId,
+      }),
+    )
+    closeUninstallConfirmationModal()
+    setIsSuccessAlertVisible(true)
+  }
+}
+
+export const onUninstallButtonClick = (
+  appId: string,
+  clientId: string,
+  installationId: string,
+  dispatch: Dispatch<any>,
+  setIsSuccessAlertVisible: (isVisible: boolean) => void,
+  closeUninstallConfirmationModal: () => void,
+) => {
+  return () => {
+    dispatch(
+      appInstallationsRequestUninstall({
+        appId,
+        installationId,
+        terminatedReason: 'User uninstall',
+        callback: handleUninstallAppSuccessCallback(
+          appId,
+          clientId,
+          dispatch,
+          setIsSuccessAlertVisible,
+          closeUninstallConfirmationModal,
+        ),
+      }),
+    )
+  }
+}
+
+export const handleSuccessAlertButtonClick = history => {
```

Comment:
This history.replace function is used a lot. Feels like we should use a util helper. Also, history is type History :-)

Reply:
We can't build a util helper for history.replace and call it directly in functions like `handleSuccessAlertButtonClick` because `useHistory` hook can only called in a function component